packages/notifier/src/types.js

```diff
@@ -101,20 +103,29 @@
  */
 
 /**
- * @typedef {any} SubscriptionInternals Purposely opaque. Will be shared between
- * machines, so it must be safe to expose. But other software should avoid
- * depending on its internal structure.
+ * @template T
+ * @typedef {Object} SubscriptionInternals Will be shared between machines, so
+ * it must be safe to expose. But other software should avoid depending on its
+ * internal structure.
+ * @property {ERef<IteratorResult<T, T>>} _head
+ * @property {Promise<SubscriptionInternals<T>>} _tail
```

Comment:
I purposely tried to make `SubscriptionInternals` opaque outside the package. Underbar is not a satisfying way to do that. Can we just give them normal names internally, but not declare them externally?

Reply:
I tried to minimise "type goo" by giving them external declarations because I wanted a way to convey the type parameter (`T`) of the notifier or subscription all the way through the sharable internals.

If you really want an opaque type, I can provide that, but it will cost some explicit casting in order to convey the parameter.